File: dmn-test-cases/standard/proto/1.1/0004-lending/translator/expected/proto3/kotlin/dmn/type/MonthlyImpl.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinition.ftl", "Monthly"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
class MonthlyImpl : Monthly {
    @get:com.fasterxml.jackson.annotation.JsonGetter("Income")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Income")
    override var income: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Expenses")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Expenses")
    override var expenses: kotlin.Number? = null

    @get:com.fasterxml.jackson.annotation.JsonGetter("Repayments")
    @set:com.fasterxml.jackson.annotation.JsonGetter("Repayments")
    override var repayments: kotlin.Number? = null

    constructor() {
    }

    constructor (expenses: kotlin.Number?, income: kotlin.Number?, repayments: kotlin.Number?) {
        this.expenses = expenses
        this.income = income
        this.repayments = repayments
    }

    override fun equals(other: Any?): Boolean {
        return equalTo(other)
    }

    override fun hashCode(): Int {
        return hash()
    }

    @Override
    override fun toString(): String {
        return asString()
    }
}

File: dmn-test-cases/standard/tck/1.1/cl3/0004-lending/translator/expected/python/dmn/type_/MonthlyImpl.py
import typing
import decimal
import datetime
import time
import isodate

import type_.Monthly


# Generated(value = {"itemDefinition.ftl", "Monthly"})
class MonthlyImpl(type_.Monthly.Monthly):
    def __init__(self, expenses: typing.Optional[decimal.Decimal] = None, income: typing.Optional[decimal.Decimal] = None, repayments: typing.Optional[decimal.Decimal] = None):
        type_.Monthly.Monthly.__init__(self)

        self.expenses = expenses
        self.income = income
        self.repayments = repayments

    def __eq__(self, other: typing.Any) -> bool:
        return self.equalTo(other)

    def __hash__(self):
        return self.hashCode()

    def __str__(self) -> str:
        return self.asString()
